node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -1679,43 +1687,28 @@ mod tests {
             )
             .unwrap();
 
-        let (reply, public_key, node_addr) = match counter_debut {
-            GossipAcceptanceResult::Reply(ref reply, ref public_key, ref node_addr) => {
-                (reply, public_key, node_addr)
-            }
+        let (debut_reply, dest_public_key, dest_node_addr) = match counter_debut {
+            GossipAcceptanceResult::Reply(
+                ref debut_reply,
+                ref dest_public_key,
+                ref dest_node_addr,
+            ) => (debut_reply, dest_public_key, dest_node_addr),
             x => panic!("Expected Reply, got {:?}", x),
         };
 
-        assert_eq!(public_key, new_debutant.public_key());
-        assert_eq!(node_addr, &new_debutant.node_addr_opt().unwrap());
+        assert_eq!(dest_public_key, new_debutant.public_key());
+        assert_eq!(dest_node_addr, &new_debutant.node_addr_opt().unwrap());
         root_node
             .add_half_neighbor_key(new_debutant.public_key().clone())
             .unwrap();
         assert_eq!(
-            GossipAcceptanceResult::Reply(reply.clone(), public_key.clone(), node_addr.clone()),
-            counter_debut
-        );
-        //TODO assert on GossipNodeRecord
-        // assert_node_records_eq(reference_node, &debut_node_two, before, after);
-        // dest_db.add_node(debuted_node_three.clone()).unwrap();
-        // dest_db.add_node(debuted_node_four.clone()).unwrap();
-        // dest_db.add_node(debuted_node_five.clone()).unwrap();
-        // dest_db.add_arbitrary_full_neighbor(root_node.public_key(), half_debuted_node.public_key());
-        // dest_db.add_arbitrary_full_neighbor(
-        //     half_debuted_node.public_key(),
-        //     debuted_node_three.public_key(),
-        // );
-        // dest_db.add_arbitrary_full_neighbor(
-        //     half_debuted_node.public_key(),
-        //     debuted_node_four.public_key(),
-        // );
-        // dest_db.add_arbitrary_full_neighbor(
-        //     half_debuted_node.public_key(),
-        //     debuted_node_five.public_key(),
-        // );
-        // let agr_man = &AccessibleGossipRecord::from(&new_debutant);
-        // let more_apropriate_neighbor =
-        //     DebutHandler::find_more_appropriate_neighbor(&subject, &dest_db, agr_man);
+            counter_debut,
+            GossipAcceptanceResult::Reply(
+                debut_reply.clone(),
+                dest_public_key.clone(),
+                dest_node_addr.clone()
```

Comment:
All these three .clone() calls above have no sound implication. They should be removed.

Reply:
this is lack of `Copy` trait for `Gossip_0v1`, which leads down to the rabbit hole for all members of `GossipNodeRecord`, do we really want to implement the `Copy` trait so much in depth just for this one test?